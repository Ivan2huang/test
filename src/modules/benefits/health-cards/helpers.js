/* eslint-disable import/prefer-default-export */
export const getQRCodeXML = ({
  fullName,
  membershipNumber,
  certificateNumber,
  insurerCode,
  insurerName,
  insurerType,
  policyNumber,
  expiryDate,
}) =>
  [
    '<?xml version="1.0"?>',
    '<medical_card>',
    '<Card_type>',
    insurerType,
    '</Card_type>',
    '<insurance_company_code>',
    insurerCode,
    '</insurance_company_code>',
    '<insurance_company_name>',
    insurerName,
    '</insurance_company_name>',
    '<insured_name>',
    fullName,
    '</insured_name>',
    '<policy_no>',
    policyNumber,
    '</policy_no>',
    '<certificate_no>',
    certificateNumber,
    '</certificate_no>',
    '<membership_no>',
    membershipNumber,
    '</membership_no>',
    '<expiry_date>',
    expiryDate,
    '</expiry_date>',
    '</medical_card>',
  ].join('');
