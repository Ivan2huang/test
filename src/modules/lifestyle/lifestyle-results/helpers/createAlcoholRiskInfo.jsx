import React from 'react';

import { AlcoholIcon, NoAlcoholIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createAlcoholRiskInfo = (intl, theme, alcoholRisk) => {
  const alcoholRiskMapping = {
    None: `${theme.lifeStyleResultOK}`,
    RegularDrinker: `${theme.lifeStyleResultWarning}`,
    HeavyDrinker: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'alcohol',
    color: alcoholRiskMapping[alcoholRisk],
    CardIcon:
      alcoholRisk === 'None' ? (
        <NoAlcoholIcon />
      ) : (
        <AlcoholIcon color={alcoholRiskMapping[alcoholRisk]} />
      ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.alcoholRiskCard.title',
      'Alcohol',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.subTitle.${alcoholRisk}`,
      alcoholRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.description.${alcoholRisk}`,
      alcoholRisk,
    ),
  };
};

export default createAlcoholRiskInfo;
