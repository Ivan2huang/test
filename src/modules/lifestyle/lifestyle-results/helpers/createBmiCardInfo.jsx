import React from 'react';

import Typography from '../../../../uiComponents/Typography';
import { formatMessage } from '../../../../helpers/helpers';

export const formatBmiScore = bmiScore => {
  const score = parseFloat(bmiScore);
  if (Number.isNaN(score)) {
    return '';
  }
  return (+bmiScore).toFixed(1);
};

const createBmiCardInfo = (intl, theme, bmiScore, bmiClass) => {
  const bmiColorMapping = {
    Healthy: `${theme.lifeStyleResultOK}`,
    Overweight: `${theme.lifeStyleResultWarning}`,
    Obese: `${theme.lifeStyleResultAlarm}`,
    Underweight: `${theme.lifeStyleResultWarning}`,
  };

  return {
    id: 'bmi',
    color: bmiColorMapping[bmiClass],
    CardIcon: (
      <Typography type="style-1" color={theme.grey3} fontWeight="bold">
        {formatBmiScore(bmiScore)}
      </Typography>
    ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.bmiCard.title',
      'BMI',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.subTitle.${bmiClass}`,
      bmiClass,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.description.${bmiClass}`,
      bmiClass,
    ),
  };
};

export default createBmiCardInfo;
