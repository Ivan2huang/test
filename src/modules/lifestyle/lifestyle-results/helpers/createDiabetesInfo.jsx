import React from 'react';

import { HeartIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createDiabetesInfo = (intl, theme, diabetesRisk) => {
  const alcoholRiskMapping = {
    Low: `${theme.lifeStyleResultOK}`,
    High: `${theme.lifeStyleResultWarning}`,
    VeryHigh: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'diabetes',
    color: alcoholRiskMapping[diabetesRisk],
    CardIcon: <HeartIcon color={alcoholRiskMapping[diabetesRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.diabetesCard.title',
      'Diabetes',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.subTitle.${diabetesRisk}`,
      diabetesRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.description.${diabetesRisk}`,
      diabetesRisk,
    ),
  };
};

export default createDiabetesInfo;
