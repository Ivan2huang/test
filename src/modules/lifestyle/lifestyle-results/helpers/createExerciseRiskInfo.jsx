import React from 'react';

import { ExerciseIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createExerciseRiskInfo = (intl, theme, exerciseRisk) => {
  const exerciseRiskMapping = {
    High: `${theme.lifeStyleResultOK}`,
    LowToModerate: `${theme.lifeStyleResultWarning}`,
    VeryLow: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'exercise',
    color: exerciseRiskMapping[exerciseRisk],
    CardIcon: <ExerciseIcon color={exerciseRiskMapping[exerciseRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.exerciseRiskCard.title',
      'Exercise',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.subTitle.${exerciseRisk}`,
      exerciseRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.description.${exerciseRisk}`,
      exerciseRisk,
    ),
  };
};

export default createExerciseRiskInfo;
