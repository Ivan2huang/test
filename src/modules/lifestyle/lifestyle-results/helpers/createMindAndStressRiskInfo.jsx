import React from 'react';

import { StressIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createMindAndStressRiskInfo = (intl, theme, mindAndStressRisk) => {
  const mindAndStressRiskMapping = {
    Normal: `${theme.lifeStyleResultOK}`,
    BeCareful: `${theme.lifeStyleResultWarning}`,
    AtRisk: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'mindAndStress',
    color: mindAndStressRiskMapping[mindAndStressRisk],
    CardIcon: (
      <StressIcon color={mindAndStressRiskMapping[mindAndStressRisk]} />
    ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.mindAndStressRiskCard.title',
      'Mind & stress',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.mindAndStressRiskCard.subTitle.${mindAndStressRisk}`,
      mindAndStressRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.mindAndStressRiskCard.description.${mindAndStressRisk}`,
      mindAndStressRisk,
    ),
  };
};

export default createMindAndStressRiskInfo;
