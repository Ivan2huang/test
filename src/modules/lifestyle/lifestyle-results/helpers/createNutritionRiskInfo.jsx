import React from 'react';

import { NutritionIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createNutritionRiskInfo = (intl, theme, nutritionRisk) => {
  const nutritionRiskMapping = {
    Low: `${theme.lifeStyleResultOK}`,
    Moderate: `${theme.lifeStyleResultWarning}`,
    High: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'nutrition',
    color: nutritionRiskMapping[nutritionRisk],
    CardIcon: <NutritionIcon color={nutritionRiskMapping[nutritionRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.nutritionRiskCard.title',
      'Nutrition',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.nutritionRiskCard.subTitle.${nutritionRisk}`,
      nutritionRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.nutritionRiskCard.description.${nutritionRisk}`,
      nutritionRisk,
    ),
  };
};

export default createNutritionRiskInfo;
