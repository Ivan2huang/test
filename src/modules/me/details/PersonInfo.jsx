import React from 'react';
import PropTypes from 'prop-types';

import { Card, Box, Button, withStyles } from '@material-ui/core';
import Typography from '../../../uiComponents/Typography';
import Grid from '../../../uiComponents/Grid';
import GridItem from '../../../uiComponents/GridItem';
import PersonInfoField from './PersonInfoField';
import { navigateTo } from '../../../helpers/helpers';
import paths from '../../../helpers/paths';

const Styles = theme => ({
  container: {
    [theme.breakpoints.down('sm')]: {
      background: 'none',
      boxShadow: 'unset',
      borderBottom: `1px solid ${theme.grey1}`,
    },
  },
  info: {
    [theme.breakpoints.down('sm')]: {
      borderBottom: `1px solid ${theme.grey1}`,
    },
  },
});

const PersonInfo = ({
  classes,
  fields,
  showInviteBtn,
  memberId,
  buttonLabel,
  isDisabled,
  invalidMessage,
}) => {
  return (
    <Box mt={2}>
      <Card classes={{ root: classes.container }}>
        <Box my={{ xs: 0, md: 8 }} mx={{ xs: 0, md: 12 }}>
          <Grid>
            {fields.map(({ label, value, editable, addable, url }) => (
              <PersonInfoField
                key={label}
                title={label}
                description={value}
                editable={editable}
                addable={addable}
                url={url}
              />
            ))}

            {showInviteBtn && (
              <GridItem columns={{ xs: 12, sm: 12, md: 12 }}>
                <Box pt={{ xs: 6, md: 6 }} pb={{ xs: 8, md: 2 }}>
                  <Button
                    variant="outlined"
                    color="secondary"
                    onClick={() =>
                      navigateTo(paths.employee.inviteDependent, {
                        memberId,
                      })
                    }
                    disabled={isDisabled}
                  >
                    {buttonLabel}
                  </Button>
                  {isDisabled && (
                    <Box pt={{ xs: 6, md: 3 }}>
                      <Typography type="style-8">{invalidMessage}</Typography>
                    </Box>
                  )}
                </Box>
              </GridItem>
            )}
          </Grid>
        </Box>
      </Card>
    </Box>
  );
};

PersonInfo.propTypes = {
  classes: PropTypes.exact({
    container: PropTypes.string.isRequired,
    info: PropTypes.string.isRequired,
  }).isRequired,
  fields: PropTypes.arrayOf(PropTypes.object),
  buttonLabel: PropTypes.string,
  invalidMessage: PropTypes.string,
  showInviteBtn: PropTypes.bool,
  isDisabled: PropTypes.bool,
  memberId: PropTypes.string,
};

PersonInfo.defaultProps = {
  memberId: '',
  invalidMessage: '',
  fields: [],
  buttonLabel: '',
  showInviteBtn: false,
  isDisabled: false,
};

export default withStyles(Styles)(PersonInfo);
