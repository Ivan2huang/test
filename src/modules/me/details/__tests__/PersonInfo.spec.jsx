/* eslint-disable react/prop-types */

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PersonInfo from '../PersonInfo';
import withTheme from '../../../../themes/withThemeProvider';
import withIntl from '../../../../i18n/withIntlProvider';
import { navigateTo } from '../../../../helpers/helpers';

jest.mock('../../../../helpers/helpers', () => ({
  navigateTo: jest.fn(),
}));

jest.mock('../../../../helpers/mui', () => ({
  useBreakpoint: jest.fn(() => 'xs'),
}));

jest.mock(
  '../../../../uiComponents/Typography',
  () => ({ children, ...rest }) => <div {...rest}>{children}</div>,
);

describe('Person Info Component', () => {
  it('should display invited text below button', () => {
    const props = {
      fields: [],
      showInviteBtn: true,
      memberId: '2',
      isDisabled: true,
      buttonLabel: 'You have successfully invited him/her.',
    };
    const Component = withIntl(withTheme(PersonInfo));
    const { getByText } = render(<Component {...props} />);

    const Text = getByText(/You have successfully invited him\/her./);

    expect(Text).toBeDefined();
  });

  it('should display invalid message', () => {
    const props = {
      showInviteBtn: true,
      memberId: '2',
      isDisabled: true,
      buttonLabel: 'You have successfully invited him/her.',
      fields: [
        {
          label: 'Name',
          value: 'Bob Tan',
          editable: false,
          addable: false,
        },
      ],
    };
    const Component = withIntl(withTheme(PersonInfo));
    const { getByText } = render(<Component {...props} />);

    const Text = getByText(/You have successfully invited him\/her./);

    expect(Text).toBeDefined();
  });

  it('should match the snapshot', () => {
    const props = {
      showInviteBtn: true,
      memberId: '2',
      isDisabled: true,
      buttonLabel: 'You have successfully invited him/her.',
      fields: [
        {
          label: 'Name',
          value: 'Bob Tan',
          editable: false,
          addable: false,
        },
      ],
    };
    const Component = withIntl(withTheme(PersonInfo));
    const { container } = render(<Component {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot for dependent', () => {
    const props = {
      fullName: 'dummyUser',
      membershipNumber: '123',
      relationship: 'Spouse',
      hasLoggedIn: true,
      fields: [
        {
          label: 'Name',
          value: 'Bob Tan',
          editable: false,
          addable: false,
        },
      ],
    };
    const Component = withIntl(withTheme(PersonInfo));
    const { container } = render(<Component {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should navigate to invite page when invite button clicked', () => {
    const props = {
      memberId: '100',
      buttonLabel: 'Invite to Benefit',
      isDisabled: false,
      showInviteBtn: true,
      fields: [
        {
          label: 'Name',
          value: 'Bob Tan',
          editable: false,
          addable: false,
        },
      ],
    };

    const Component = withIntl(withTheme(PersonInfo));
    const { getByText } = render(<Component {...props} />);

    const InviteButton = getByText(/Invite to/);
    fireEvent.click(InviteButton, {});

    expect(navigateTo).toHaveBeenCalledWith('/me/details/invite', {
      memberId: '100',
    });
  });
});
