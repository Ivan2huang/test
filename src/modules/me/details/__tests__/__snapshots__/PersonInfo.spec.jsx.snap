// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Info Component should match the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-321"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root PersonInfo-container-318 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-350"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-445"
            >
              <div
                type="style-6"
              >
                Name
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiBox-root MuiBox-root-446"
                >
                  <div
                    color="highEmphasis"
                    type="style-5"
                  >
                    Bob Tan
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-447"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  You have successfully invited him/her.
                </span>
              </button>
              <div
                class="MuiBox-root MuiBox-root-479"
              >
                <div
                  type="style-8"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Person Info Component should match the snapshot for dependent 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-483"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root PersonInfo-container-480 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-512"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-607"
            >
              <div
                type="style-6"
              >
                Name
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiBox-root MuiBox-root-608"
                >
                  <div
                    color="highEmphasis"
                    type="style-5"
                  >
                    Bob Tan
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
